src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.cs

```diff
@@ -178,7 +192,10 @@ async ValueTask<(Solution? solution, T result)> TryFastGetSolutionAndRunAsync()
             // We use a reference-counted solution that implements IAsyncDisposable. The computation of 'newSolution'
             // uses eager cancellation, but the asynchronous disposable applies lazy cancellation to the final task that
             // causes cancellation to propagate to the backing lazy operation.
-            var refCountedLazySolution = await GetLazySolutionAsync().ConfigureAwait(false);
+            var refCountedLazySolution = await GetLazySolutionAsync(
+                solutionChecksum,
+                cancellationToken => ComputeSolutionAsync(assetProvider, solutionChecksum, currentSolution, cancellationToken),
```

Comment:
the two calls to GetAndCacheLazySolutionAsync are parameterized on how they create the solution.  in teh slow path here we actually have to go through all the  work of syncing/computing if not in the cache.  IN the fast path far above, we can just jam in the value directly.